nerfstudio/data/dataparsers/nerfstudio_dataparser.py

```diff
@@ -87,6 +88,10 @@ def _generate_dataparser_outputs(self, split="train"):
             else:
                 image_filenames.append(fname)
                 poses.append(np.array(frame["transform_matrix"]))
+            if "mask_path" in frame:
+                mask_filepath = PurePath(frame["mask_path"])
+                mask_fname = self._get_fname(mask_filepath)
```

Comment:
So I don't think `self._get_fname` will work if downscale_factor != 1. This is specific for the downscaled RGB images. Were you able to test this?

Reply:
Right, agreed with Matt. Instead of an "images" folder, it can be called "masks".

Reply:
Added the option to specify the folder prefix to `_get_fname` - The masks will be searched for in `masks_X`, where `X` is the downscale level.

Reply:
Another option here (I do this in my local fork for custom data) is to support image / mask / semantic downscaling in the dataset. It is common to only store images at a single resolution so probably makes sense to support that use case?

Reply:
I like that idea a lot @nikmo33 - In my opinion it's beyond the scope of this PR, though.